File: StackQueue/bridge.py
def solution(bridge_length, weight, truck_weights):
    bridge = []
    for i in range(bridge_length):
        bridge.append(0)
    count = 0
    cur_weight = 0
    while truck_weights:
        cur_weight -= bridge[0]
        bridge.pop(0)
        if cur_weight + truck_weights[0] <= weight:
            cur_weight += truck_weights[0]
            bridge.append(truck_weights[0])
            truck_weights.pop(0)
        else:
            bridge.append(0)
        count += 1
    return count + bridge_length

print(solution(2, 10, [7,4,5,6]))

File: Programmers/StackQueue/bridge.cpp
#include <iostream>
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(int bridge_length, int weight, vector<int> truck_weights)
{
    queue<int> bridge;
    for (int i = 0; i < bridge_length; i++)
        bridge.push(0);
    vector<int> temp = vector<int>(truck_weights.rbegin(), truck_weights.rend());
    int count = 0, current_weight = 0;
    while (!temp.empty())
    {
        current_weight -= bridge.front();
        bridge.pop();
        if (current_weight + temp.back() <= weight)
        {
            bridge.push(temp.back());
            current_weight += temp.back();
            temp.pop_back();
        }
        else
        {
            bridge.push(0);
        }
        count++;
    }
    return count + bridge_length;
}

int main()
{
}